src/runtime/startup/c/wrapper.c

```diff
@@ -519,6 +527,17 @@ __attribute__((constructor)) static void init(void) {
 
     config.isSuid = is_suid();
 
+    if ( config.isSuid || geteuid() == 0 ) {
+        /* force kernel to load overlay module to ease detection later */
+        if ( mount("none", "/", "overlay", MS_SILENT, "") < 0 ) {
+            if ( errno != EINVAL ) {
+                singularity_message(DEBUG, "Overlay seems not supported by kernel\n");
```

Comment:
Should we do something else besides just print a debug message if not supported? Or just continue on and later if we *need* overlay then we can fail

Reply:
Nothing to do here, just a trick to force overlay module to be loaded for further detection in Go code with proc helper :)

Reply:
OH... I see what you've done there. That's sneaky and awesome trick :)